manual/tools/texquote2.ml

```diff
@@ -0,0 +1,143 @@
+type environment =
+  | Caml
+  | Verbatim
+  | Verbatim_like
+
+let (current_environment : environment option ref) = ref None
+
+let in_quotes = ref false
+
+let verbatim_end_in = ref ""
+
+let verbatim_end_out = ref ""
+
+let is_alpha c =
+  ('A' <= c && c <= 'Z') || ('a' <= c && c <= 'z')
+
+let is_prefix prefix str =
+  let length_prefix = String.length prefix in
+  let length_str = String.length str in
+  if length_prefix > length_str
+  then false
+  else (String.sub str 0 length_prefix) = prefix
+
+let escape = function
+  | ' ' | '\n' -> "\\ "
+  | '{' -> "{\\char123}"
+  | '}' -> "{\\char125}"
+  | '^' -> "{\\char94}"
+  | '_' -> "{\\char95}"
+  | '\\' -> "{\\char92}"
+  | '~' -> "{\\char126}"
+  | '$' -> "\\$"
+  | '&' -> "{\\char38}"
+  | '#' -> "\\#"
+  | '%' -> "\\%"
+  | '\'' -> "{\\textquotesingle}"
+  | '`' -> "{\\textasciigrave}"
+  | _ -> ""
```

Comment:
Is the catch-all case correct?

Reply:
At least it is not misused, since all uses of escape test if the escaped string is `""` before printing either the original character or the expansion. Nevertheless, I think that replacing `escape` by `print_escaped` would remove any possibility of misuse.